### PR TITLE
Refactor `databricks_external_location` to Go SDK

### LINKED ISSUE #2540: [ISSUE] Issue with `databricks_external_location` resource: fails on storage url update in Azure
Hello, i have noticed an issue with 'databricks_external_location', to be exact, when i recreate Storage Account with a new name, Terraform updates in-place 'url' parameter of 'databricks_external_location'. During apply it destroys current and creates new Storage Account with new name and then tries to update existing 'databricks_external_location' with url that leads to new Storage Account. However, while updating Terraform fails. 

Maybe it would be better idea to force recreate of storage location when url updated?
### Configuration
```hcl
variable "storage_account_name" {}

resource "databricks_external_location" "some" {
  name  = "example"
  url  = "abfss://container@${var.storage_account_name}.dfs.core.windows.net"
  credential_name = databricks_storage_credential.external.id
}
```

### Expected Behavior

Successful update of External Location with url to new Storage Account.

### Actual Behavior

To update External Location it is required to recreate it.

image

image

### Steps to Reproduce
1. `terraform apply -var="storage_account_name=example1"`
2.`terraform apply -var="storage_account_name=example2"`

### Terraform and provider versions
```
Terraform v1.5.2
on darwin_arm64
+ provider registry.terraform.io/databricks/databricks v1.21.0
+ provider registry.terraform.io/hashicorp/azurerm v3.64.0
+ provider registry.terraform.io/hashicorp/http v3.4.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.5.1
+ provider registry.terraform.io/microsoft/azuredevops v0.7.0
```

### LINKED ISSUE #2545: [ISSUE] Issue with `databricks_XXX` resource
Cx [Ask Locala] tried updating the URL of some external locations using terraform but it does not work.

Example after plan :

Terraform will perform the following actions:

# databricks_external_location.external_location_UC["r4u-lake-displayce-bronze-campaigns_hourly-dev"] will be updated in-place
~ resource "databricks_external_location" "external_location_UC" {
id = "r4u-lake-displayce-bronze-campaigns_hourly-dev"
name = "r4u-lake-displayce-bronze-campaigns_hourly-dev"
~ url = "s3://r4u-lake-displayce-dev/data/bronze/campaigns_hourly/delta" -> "s3://r4u-lake-displayce-dev/data/bronze/campaigns_hourly"
# (7 unchanged attributes hidden)
}

Plan: 0 to add, 1 to change, 0 to destroy.

^ here it says it's going to update the URL, this is good.

When applying, get the following :

Error: cannot update external location: External Location has dependent catalogs, schemas, managed or external tables/volumes; use force option to update url anyway
│
│ with databricks_external_location.external_location_UC["r4u-lake-displayce-bronze-campaigns_hourly-dev"],
│ on databricks_external_location.tf line 21, in resource "databricks_external_location" "external_location_UC":
│ 21: resource "databricks_external_location" "external_location_UC" {
...

Even if we have the following in the resource :

resource "databricks_external_location" "external_location_UC" {
for_each = local.external_locations_UC
name = each.key
url = each.value
credential_name = databricks_storage_credential.databricks-unity-catalog_external.id
comment = "Managed by TF"
force_destroy = true
skip_validation = true
}

We tried using SQL, doing :

%sql

ALTER EXTERNAL LOCATION `r4u-lake-displayce-bronze-campaigns_hourly-dev`
SET URL 's3://r4u-lake-displayce-dev/data/bronze/campaigns_hourly' FORCE;

And this time it's working so it looks like it's a API/terraform problem

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes

- Refactor `databricks_external_location` to Go SDK
- Add support for AccessPoint and EncryptionDetails (for SSE) - supersedes #2539
- Add `force_update` parameter. Close #2545 , close #2540

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

**Comment by adamcain-db** [comment]
@nkvuong : will the test failures be fixed now that https://github.com/databricks/universe/pull/357111 is merged?

**Review by alexott — APPROVED** [review]
lgtm

File: catalog/resource_external_location.go [hunk 1]
```diff
@@ -2,45 +2,36 @@ package catalog
...
+	AwsKmsKeyArn string `json:"aws_kms_key_arn,omitempty"`
 }
 
-func (a ExternalLocationsAPI) get(name string) (el ExternalLocationInfo, err error) {
-	err = a.client.Get(a.context, "/unity-catalog/external-locations/"+url.PathEscape(name), nil, &el)
-	return
+type EncryptionDetails struct {
+	// This is a oneOf type, but only one type defined currently:
+	SseEncDetails *SseEncryptionDetails `json:"sse_encryption_details,omitempty"`
 }
```
Comment (adamcain-db) [inline catalog/resource_external_location.go:22]: Should use the Go SDK types, right?  See [this PR](https://github.com/databricks/terraform-provider-databricks/pull/2539/files#diff-0920ed275cfdb0ac2f823d9380fcfe2cbba9c0622c8f487836f1f79a6f5a06c5).

**Review by mgyucht — APPROVED** [review]
LGTM!

**Pull request merged**